openmc/bounding_box.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+from .checkvalue import check_type, check_length
+
+
+class BoundingBox(tuple):
+    """Axis-aligned bounding box.
+
+    Parameters
+    ----------
+    corners : 2-tuple of numpy.array
+        Lower-left and upper-right coordinates of an axis-aligned bounding box
+        of the domain.
+
+    Attributes
+    ----------
+    center: numpy.array
+        x, y, z coordinates of the center of the bounding box in cm.
+    lower_left: numpy.array
+        The x, y, z coordinates of the lower left corner of the bounding box
+    upper_right
+        The x, y, z coordinates of the upper right corner of the bounding box
+    volume: float
+        The volume of the bounding box in cm3
+    """
+
+    def __init__(self, corners):
+
+        check_type("corners", corners, tuple)
+        check_type("corners", corners[0], np.ndarray)
```

Comment:
Requiring this to be a tuple of NumPy array feels a little restrictive. Creating a bounding box from scratch currently requires a line like this:

```python
bbox = openmc.BoundingBox((np.asarray([0.0, 0.0, 0.0]), np.asarray([1.0, 1.0, 1.0])))
```

whereas with the allowance of general iterables of floats this reduces to:

```python
bbox = openmc.BoundingBox([[0.0, 0.0, 0.0], [1.0, 1.0, 1.0]])
```

which I personally find a little easier to work with.


What would you say to converting the arguments to `np.ndarray` as needed with `np.asarray`?